File: cpp/1109.corporate-flight-bookings.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
	    vector<int> res(n);
	    vector<int> counter(n+2);
	    for(auto book:bookings){
		    counter[book[0]]+=book[2];
		    counter[book[1]+1]-=book[2];
	    }

	    int current = 0 ;
	    for(int i = 0 ; i < n ;i++){
		    current += counter[i+1];
		    res[i] = current;
	    }
	    return res;
    }
};

File: python/1109.corporate-flight-bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        prefix_sum = [0]*(n+2)
        for book in bookings:
            prefix_sum[book[0]]+=book[2]
            prefix_sum[book[1]+1] -= book[2]
        current_sum = 0
        res = []
        for i in range(1, n+1):
            current_sum += prefix_sum[i]
            res.append(current_sum)
        return res
